### PR TITLE
New Keyboard: Orange 75 BMC 32a version

### PR DESCRIPTION
**Pull request by mechmerlin**
## Description

This PR not only introduces a new 32a PCB in to the orange75 mix, but also puts the existing Orange75 into the foxlab directory. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/foxlab/orange75/32a/readme.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+* Keyboard Maintainer: [MechMerlin](https://www.github.com/mechmerlin)
+* Hardware Supported: Orange75 powered by the ATmega32a 
+* Hardware Availability: [zFrontier](https://en.zfrontier.com/collections/keyboards/products/foxlab-orange75), [Geekhack GB](https://geekhack.org/index.php?topic=95080.0)
+
+
+Make example for this keyboard (after setting up your build environment):
+
+    make foxlab/orange75/32a:default
+
+**Reset Key**: Hold down the key located at *LOCATION*, commonly programmed as *KEY* while plugging in the keyboard.
```
Comment (noroadsleft) [inline keyboards/foxlab/orange75/32a/readme.md]: `s/*LOCATION*/???`
`s/*KEY*/???`

File: keyboards/foxlab/orange75/32a/32a.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    matrix_scan_user();
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    // put your per-action keyboard code here
+    // runs for every action, just before processing by the firmware
+
+    return process_record_user(keycode, record);
+}
```
Comment (fauxpark) [inline keyboards/foxlab/orange75/32a/32a.c]: These functions can be deleted.

File: keyboards/foxlab/orange75/32a/keymaps/default/config.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2019 MechMerlin
```
Comment (fauxpark) [inline keyboards/foxlab/orange75/32a/keymaps/default/config.h]: This file can be deleted.

File: keyboards/foxlab/orange75/32a/usbconfig.h [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+ * USB-IDs-for-free.txt for details!
+ * *** IMPORTANT NOTE ***
+ * This template uses obdev's shared VID/PID pair for Vendor Class devices
+ * with libusb: 0x16c0/0x5dc.  Use this VID/PID pair ONLY if you understand
+ * the implications!
+ */
+#define USB_CFG_DEVICE_VERSION (DEVICE_VER & 0xFF), ((DEVICE_VER >> 8) & 0xFF)
+/* Version number of the device: Minor number first, then major number.
+ */
+#define USB_CFG_VENDOR_NAME 'w', 'i', 'n', 'k', 'e', 'y', 'l', 'e', 's', 's', '.', 'k', 'r'
```
Comment (fauxpark) [inline keyboards/foxlab/orange75/32a/usbconfig.h]:
```suggestion
#define USB_CFG_VENDOR_NAME 'F', 'o', 'x', '-', 'L', 'a', 'b'
```

File: keyboards/foxlab/orange75/32a/usbconfig.h [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+#define USB_CFG_VENDOR_NAME_LEN 13
```
Comment (fauxpark) [inline keyboards/foxlab/orange75/32a/usbconfig.h]:
```suggestion
#define USB_CFG_VENDOR_NAME_LEN 7
```

File: keyboards/foxlab/orange75/32a/usbconfig.h [hunk 6]
```diff
@@ -0,0 +1,373 @@
...
+/* These two values define the vendor name returned by the USB device. The name
+ * must be given as a list of characters under single quotes. The characters
+ * are interpreted as Unicode (UTF-16) entities.
+ * If you don't want a vendor name string, undefine these macros.
+ * ALWAYS define a vendor name containing your Internet domain name if you use
+ * obdev's free shared VID/PID pair. See the file USB-IDs-for-free.txt for
+ * details.
+ */
+#define USB_CFG_DEVICE_NAME 'p', 's', '2', 'a', 'v', 'r', 'G', 'B'
```
Comment (fauxpark) [inline keyboards/foxlab/orange75/32a/usbconfig.h]:
```suggestion
#define USB_CFG_DEVICE_NAME 'O', 'r', 'a', 'n', 'g', 'e', '7', '5'
```

File: keyboards/foxlab/orange75/32a/usbconfig.h (hunk 4 above)
Comment (mechmerlin) [reply keyboards/foxlab/orange75/32a/usbconfig.h]: editing this causes BMC to no longer work with these boards. I'd like to preserve backward compatibility.

File: keyboards/foxlab/orange75/32a/usbconfig.h (hunk 5 above)
Comment (mechmerlin) [reply keyboards/foxlab/orange75/32a/usbconfig.h]: editing this causes BMC to no longer work with these boards. I'd like to preserve backward compatibility.

File: keyboards/foxlab/orange75/32a/usbconfig.h (hunk 6 above)
Comment (mechmerlin) [reply keyboards/foxlab/orange75/32a/usbconfig.h]: editing this causes BMC to no longer work with these boards. I'd like to preserve backward compatibility.

File: keyboards/foxlab/orange75/32a/usbconfig.h (hunk 4 above)
Comment (fauxpark) [reply keyboards/foxlab/orange75/32a/usbconfig.h]: It already won't work due to this: https://github.com/qmk/qmk_firmware/pull/7456

Comment (mechmerlin) [reply keyboards/foxlab/orange75/32a/usbconfig.h]: BMC will not and never has worked with QMK. What this does is, even if you were to flash on the ps2avrgb firmware, BMC would still not recognize the board.

Comment (zvecr) [reply keyboards/foxlab/orange75/32a/usbconfig.h]: How does the vendor name from a previous firmware flash cause issues? Is windows caching something?

File: keyboards/foxlab/orange75/32a/config.h [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+//                         0   1   2   3   4   5   6   7   8   9   A   B   C   D  E
+#define MATRIX_ROW_PINS { B5, B0, B1, B2, B3, B4 }
+#define MATRIX_COL_PINS { C2, C4, C5, C6, C7, A7, A6, A5, A4, A3, A2, A1, A0, C3, D7 }
+// #define MATRIX_COL_PINS { A0, A1, A2, A3, A4, A5, A6, A7, C7, C6, C5, C4, C3, C2, C1, C0, D7 }
+#define UNUSED_PINS {}
+
+#define DIODE_DIRECTION COL2ROW
+#define DEBOUNCE 5
+
+#define BACKLIGHT_LEVELS 1
```
Comment (fauxpark) [inline keyboards/foxlab/orange75/32a/config.h]: So I watched the porting stream for this board 😁 you know how you were trying to find the pins for the indicator LEDs? May want to bookmark this: https://github.com/showjean/ps2avrU/blob/master/firmware/ps2avrGB/Circuit/ps2avrGB/20140829_140848_001.png
A lot of these BMC ports seem to just treat them as on/off backlight pins, eg. `backlight_set()` in winkeyless/bface, which is incorrect.

And I can see this PCB has backlight pads, so per the schematic that'll be:

```suggestion
#define BACKLIGHT_PIN D4
#define BACKLIGHT_LEVELS 3
```
D4 is a hardware PWM pin on the 32A, too 👍

File: keyboards/foxlab/orange75/32a/usbconfig.h (hunk 4 above)
Comment (noroadsleft) [reply keyboards/foxlab/orange75/32a/usbconfig.h]: In the interest of moving this along (though I'm not mechmerlin):

My understanding is that the [Bootmapper Client app](https://blog.winkeyless.kr/155) only recognizes the board if the Vendor and Device IDs are `winkeyless.kr` and `ps2avrGB`, respectively. Any other values prevent the app from recognizing the board, which means the user is prevented from changing from QMK to BMC, should they choose to do so.

Comment (fauxpark) [reply keyboards/foxlab/orange75/32a/usbconfig.h]: The PR I linked prevents BMC from doing this automatically, even if the strings in QMK are the same. Thus, the only way to back to BMC now is to flash the PS2AVRGB hex manually, in which case the strings *will* be correct.

**Pull request closed without merging**